google/cloud/internal/pagination_range.h

```diff
@@ -29,6 +30,18 @@ namespace cloud {
 inline namespace GOOGLE_CLOUD_CPP_NS {
 namespace internal {
 
+inline bool ComparePaginationValues(google::protobuf::Message const& lhs,
+                                    google::protobuf::Message const& rhs) {
+  return google::protobuf::util::MessageDifferencer::Equals(lhs, rhs);
+}
+
+template <typename T, typename std::enable_if<
```

Comment:
(optional): I might have written this overload sfinae'ing on the existence of `==`. Maybe like:

```cc
template <typename T>
auto ComparePaginationValues(T const& a, T const& b) -> decltype(a == b) {
return a == b;
}
```
Not sure which way is really "preferred", but something for you to consider. Happy with whatever you decide.

Reply:
I will add a separate bug for that... seems intriguing, but I am happy with the code as-is for now.

Reply:
#4814